src/index.ts

```diff
@@ -195,20 +199,32 @@ export const createProxy = <T>(
   obj: T,
   affected: WeakMap<object, unknown>,
   proxyCache?: WeakMap<object, unknown>,
+  targetCache?: WeakMap<object, unknown>,
 ): T => {
   if (!isObjectToTrack(obj)) return obj;
-  const target = getOriginalObject(obj);
-  const isTargetCopied = needsToCopyTargetObject(target);
+  let targetAndCopied = (
+    targetCache && (targetCache as TargetCache<typeof obj>).get(obj)
+  );
+  if (!targetAndCopied) {
+    const target = getOriginalObject(obj);
+    if (needsToCopyTargetObject(target)) {
+      targetAndCopied = [target, copyTargetObject(target)];
+    } else {
+      targetAndCopied = [target];
+    }
+    targetCache?.set(obj, targetAndCopied);
```

Comment:
Wdyt of forcing callers to pass `targetCache` if they want the `copyTargetObject` behavior? Just so that they aren't surprised by bad perf.

I'm also naively curious if we could just have a global top-level `targetCache`.

Reply:
> Wdyt of forcing callers to pass `targetCache` if they want the `copyTargetObject` behavior? Just so that they aren't surprised by bad perf.

Well, `proxyCache` and `targetCache` are optional, and not always necessary, and can't be global.

(btw, I'm more on your suggestion about removing support of copying completely. Maybe in v2 in the future.)

> I'm also naively curious if we could just have a global top-level `targetCache`.

Been there done that. #53 It broke with getters and proxy-memoize, it's an edge case though. But, that convinced me that global cache is a bad idea.